packages/documentation/src/stories/components/input/input.snapshot.stories.ts

```diff
@@ -10,68 +11,65 @@ export default {
-      value: 'Lorem Ipsum',
-    },
-  ];
   return html`
     <div class="d-flex flex-wrap align-items-start gap-regular">
       ${['bg-white', 'bg-dark'].map(
         bg => html`
           <div class="${bg} d-flex gap-3 flex-column p-3">
             <h3>Sizes</h3>
-            ${getCombinations('size', context.argTypes.size.options, combinations)
-              .filter(
-                (args: Args) =>
-                  !args.value ||
-                  (args.value &&
-                    (context.args.type === 'text' || context.args.type === 'password')),
-              )
-              .map(
-                (args: Args) =>
-                  html`
-                    <div>
-                      ${args.title !== undefined && args.title
-                        ? html`
-                            <h4>
-                              ${Object.entries(context.argTypes.size.control.labels)
-                                .filter(([key, value]) => key === args.size)
-                                .map(s => s[1])}
-                            </h4>
-                          `
-                        : ''}
-                      <div>${meta.render?.({ ...context.args, ...args }, context)}</div>
-                    </div>
-                  `,
-              )}
+            ${context.argTypes.size.options.map((size: string) => {
+              const variants = bombArgs({
+                combinations: COMBINATIONS,
+                size: [size],
+                pseudoClass: ['null', 'focus'],
+              })
+                .map(extractCombinationsArgs)
+                .filter(
+                  (args: Args) =>
+                    !args.value ||
+                    (args.value &&
+                      (context.args.type === 'text' || context.args.type === 'password')),
+                )
+                .map(args => renderVariant(args, context));
+
+              return html`
+                <section>
+                  <h4>${context.argTypes.size.control.labels[size]}</h4>
+                  <div class="d-flex gap-3 flex-column">${variants}</div>
+                </section>
+              `;
+            })}
             <h3>Floating Label</h3>
-            ${getCombinations('floatingLabel', [true], combinations)
+            ${bombArgs({
+              combinations: COMBINATIONS,
+              floatingLabel: [true],
+              pseudoClass: ['null', 'focus'],
+            })
+              .map(extractCombinationsArgs)
               .filter(
                 (args: Args) =>
                   !args.value ||
                   (args.value &&
                     (context.args.type === 'text' || context.args.type === 'password')),
               )
-              .map(
-                (args: Args) =>
-                  html`
-                    <div>${meta.render?.({ ...context.args, ...args }, context)}</div>
-                  `,
-              )}
+              .map(args => renderVariant(args, context))}
```

Comment:
I refactored this part to use `bombArgs` instead of a custom function. So https://github.com/swisspost/design-system/issues/2001 will be easier.